docker/UCXPy-CUDA.dockerfile

```diff
@@ -0,0 +1,71 @@
+ARG CUDA_VER=11.5.2
+ARG LINUX_VER=ubuntu20.04
+# Should match host OS OFED version (as reported by ofed_info -s)
+ARG OFED_VER=5.3-1.0.5.0
+FROM nvidia/cuda:${CUDA_VER}-devel-${LINUX_VER}
+
+# Make these values available to later build stages
+ARG CUDA_VER
+ARG LINUX_VER
+ARG OFED_VER
+ENV CUDA_HOME="/usr/local/cuda"
+ENV PATH="/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends tzdata && \
+    apt-get install -y \
+        automake \
+        dh-make \
+        git \
+        libcap2 \
+        libnuma-dev \
+        libtool \
+        make \
+        pkg-config \
+        udev \
+        wget \
+    && apt-get autoremove -y \
+    && apt-get clean
+
+ADD https://content.mellanox.com/ofed/MLNX_OFED-${OFED_VER}/MLNX_OFED_LINUX-${OFED_VER}-${LINUX_VER}-x86_64.tgz /MLNX_OFED_LINUX-${OFED_VER}-${LINUX_VER}-x86_64.tgz
+RUN tar -xzf /MLNX_OFED_LINUX-${OFED_VER}-${LINUX_VER}-x86_64.tgz
+RUN cd MLNX_OFED_LINUX-${OFED_VER}-${LINUX_VER}-x86_64 && \
+ yes | ./mlnxofedinstall --user-space-only --without-fw-update --without-neohost-backend \
+ && rm -rf /var/lib/apt/lists/* \
+ && rm -rf /MLNX_OFED_LINUX*
+
+ADD https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh /miniconda.sh
+RUN bash /miniconda.sh -b -p /opt/conda
+
+ENV PATH="/opt/conda/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+# FIXME: We plausibly have two different CUDA runtimes after this.
+# The base image provides a build environment at /usr/local/cuda
+# but this install adds the cudatoolkit from the nvidia conda channel.
+# Hopefully they don't conflict...
+RUN conda create -n ucx -c conda-forge -c nvidia -c rapidsai \
+    "python=3.7" setuptools psutil "cython>=0.29.14,<3.0.0a0" \
+    pytest pytest-asyncio \
+    cupy "numba>=0.46" rmm distributed
+
+# Setup for interactive use of conda
+RUN conda init bash
+
+WORKDIR /root
+RUN git clone https://github.com/openucx/ucx.git
+WORKDIR /root/ucx
+RUN git checkout v1.12.1
+RUN ./autogen.sh
+WORKDIR /root/ucx/build
+# FIXME: how to get conda activate to actually work (hard-coded prefix here is bad)
+# conda env list --json | python -c "import json, sys; env, = (e for e in json.load(sys.stdin)['envs'] if e.endswith('ucx')); print(env)"
```

Comment:
I think what you're missing here is just:

```
source /datasets/pentschev/miniconda3/etc/profile.d/conda.sh
```

Reply:
Thanks, done.